leaf-ui/rappid-extensions/BackboneModelsIntention.js

```diff
@@ -7,61 +8,404 @@
+     * which means there is no loner a repeating function segment in the EvolvingFunctionBBM     *  
+     */ 
+    removeRepFuncSegments: function() {
+
+        hasRepeat = false;
+        repStart = null;
+        repStop = null;          
+        repCount = null;         
+        repAbsTime = null;  
+    },    
+
 });
 
+/**
+ * Backbone Model of Intention
+ * 
+ * @param {String} nodeName
+ * @param {String} nodeActorID
+ * ID of the Actor the Intention belongs to
+ * @param {String} nodeType
+ * Type of node. Ex: task, goal
+ * @param {EvolvingFunctionBBM} evolvingFunction
+ * If there is an evolving function this will contain an EvolvingFunctionBBM
+ * @param {String} initialValue
+ * String of numbers that represents the initial satisfaction value
+ */
 var IntentionBBM = Backbone.Model.extend({
+
     initialize: function(options) { 
-        _.extend({}, this.defaults, options)
+        _.extend({}, this.defaults, options) 
     }, 
+
     defaults: { 
-        nodeName: "untitled",
-        nodeType: null,
-        nodeActorID: null,                     // Assigned on release operation.
-        evolvingFunction: null, 
-        initialValue: '(no value)'
-    },
+            nodeName: 'untitled',
+            nodeActorID: null,                     // Assigned on release operation.
+            nodeType: null,
+            evolvingFunction: null, 
+            initialValue: '(no value)'
+    }, 
 
     /**
-     * @returns Array of FunctionSegmentBBMs, or null if it is not an evolvingFunction
-     */
+    * @returns Array of FunctionSegmentBBMs, or null if it is not an evolvingFunction
+    */
     getFuncSegments: function(){
+
         evolvingFunc = this.get('evolvingFunction');
         if (evolvingFunc != null){
+            console.log(evolvingFunc.get('functionSegList'));
             return evolvingFunc.get('functionSegList');
         }
         return null;
-    }
+    },
+ 
+    /**
+     * Called whenever the user changes or removes the Initial Satisfaction Value
+     * in the Element Inspector. Called on Change for #init-sat-value 
+     * 
+     * Sets the refEvidence pair based on the chosen initValue
+     * 
+     * @param {*} initValue 
+     */
+    changeInitialSatValue: function(initValue) {
+
+        var intentionEval = graph.getUserEvaluationBBM(this.cid, '0');
+        intentionEval.set('assignedEvidencePair', initValue);
+        var funcSegList = this.getFuncSegments();
+        
+        if (this.get('evolvingFunction') != null) {
+            if (this.get('evolvingFunction').get('type') == 'C' || 
+                (this.get('evolvingFunction').get('type') == 'UD' && funcSegList[0].get('type') == 'C')) { 
+                    funcSegList[0].set('refEvidencePair', initValue); // Set first index of funcSegList to given initValue 
+            }
+            this.get('evolvingFunction').set('type', 'NT');
+            this.get('evolvingFunction').set('functionSegList', []);
```

Comment:
Agreed that we need more comments here. Also, there is a bug, because if I change the value for a 'C' function, then it also resets the evolving function. Perhaps, we are missing an `else`?